analytical_engine/java/grape-demo/src/main/java/com/alibaba/graphscope/example/pagerank/PageRank.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2021 Alibaba Group Holding Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.graphscope.example.pagerank;
+
+import com.alibaba.graphscope.app.ParallelAppBase;
+import com.alibaba.graphscope.communication.Communicator;
+import com.alibaba.graphscope.context.ParallelContextBase;
+import com.alibaba.graphscope.ds.Vertex;
+import com.alibaba.graphscope.ds.VertexRange;
+import com.alibaba.graphscope.ds.adaptor.AdjList;
+import com.alibaba.graphscope.ds.adaptor.Nbr;
+import com.alibaba.graphscope.fragment.SimpleFragment;
+import com.alibaba.graphscope.parallel.ParallelEngine;
+import com.alibaba.graphscope.parallel.ParallelMessageManager;
+import com.alibaba.graphscope.parallel.message.DoubleMsg;
+import com.alibaba.graphscope.utils.FFITypeFactoryhelper;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PageRank extends Communicator
+        implements ParallelAppBase<Long, Long, Long, Double, PageRankContext>, ParallelEngine {
+    private static Logger logger = LoggerFactory.getLogger(PageRank.class);
+
+    @Override
+    public void PEval(
+            SimpleFragment<Long, Long, Long, Double> fragment,
+            ParallelContextBase<Long, Long, Long, Double> contextBase,
+            ParallelMessageManager javaParallelMessageManager) {
```

Comment:
Could it be `parallelMessageManager`? The `java` prefix looks a bit strange..

Reply:
As it is in jave code.

Reply:
No Prob.